src/sparseml/pytorch/sparsification/pruning/modifier_pruning_obs.py

```diff
@@ -518,7 +518,7 @@ def mask_update(self, masks: List[Tensor], mask_diffs: List[Tensor]):
                     obs_updates[i] = (
                         self._finvs[i]
                         .mul(
-                            self._block_finv_w[i].view(-1)
+                            self._block_finv_w[i].reshape(-1)
```

Comment:
Why is this necessary?

Reply:
Get this error on 1.13:
```python
self = <sparseml.pytorch.sparsification.pruning.modifier_pruning_obs.OBSPruningParamsScorer object at 0x7f6a443632b0>
masks = [tensor([[1., 1., 1., 1., 1., 1., 1., 1.],
        [1., 1., 1., 1., 1., 1., 1., 1.],
        [1., 1., 1., 1., 1., 1., ...,  ..., 1., 1., 1.],
        [1., 1., 1.,  ..., 1., 1., 1.],
        [1., 1., 1.,  ..., 1., 1., 1.]], device='cuda:2')]
mask_diffs = [tensor([[0., 0., 0., 0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 0., 0., ...      [0., 0., 0.,  ..., 0., 0., 0.],
        [0., 0., 0.,  ..., 0., 0., 0.],
        [0., 0., 0.,  ..., 0., 0., 0.]])]

    @torch.no_grad()
    def mask_update(self, masks: List[Tensor], mask_diffs: List[Tensor]):
        """
        Apply OBS weight update which zeros-out pruned weights and updates the
        remaining weights to preserve the loss.
    
        :param masks: latest masks to be applied to these parameters
        :param mask_diffs: mask diff values returned by mask_difference for these
            masks that describe how these masks changed since the last update
        """
        obs_updates = [None] * len(self._params)
        if self._is_main_proc:
            for i, param in enumerate(self._params):
                if self._mask_type == "unstructured":
                    obs_updates[i] = (
                        self._finvs[i]
                        .mul(
                            (param.data * (mask_diffs[i] == -1))
                            .view(-1)
                            .to(self._devices[i])
                            / (self._finvs[i].diag() + self._eps)
                        )
                        .view(param.data.shape)
                    )
                else:  # self._mask_type == "block4":
                    obs_updates[i] = (
                        self._finvs[i]
                        .mul(
>                           self._block_finv_w[i].view(-1)
                            * (mask_diffs[i] == -1).view(-1).to(self._devices[i])
                        )
                        .view(param.data.shape)
                    )
E                   RuntimeError: view size is not compatible with input tensor's size and stride (at least one dimension spans across two contiguous subspaces). Use .reshape(...) instead.

src/sparseml/pytorch/sparsification/pruning/modifier_pruning_obs.py:521: RuntimeError
```